src/idpyoidc/server/oauth2/token_helper.py

```diff
@@ -236,6 +237,103 @@ def post_parse_request(
         return request
 
 
+class CCAccessTokenHelper(TokenEndpointHelper):
+    def process_request(self, req: Union[Message, dict], **kwargs):
+        """
+        :param req:
+        :param kwargs:
+        :return:
+        """
+        _context = self.endpoint.server_get("endpoint_context")
+        _mngr = _context.session_manager
+        logger.debug("Access Token")
+
+        _token_usage_rules = _context.authz.usage_rules(req["client_id"])
+        _authn_event = create_authn_event(
+            req["client_id"]
+        )
+        sid = _mngr.create_session(
+            authn_event=_authn_event,
+            auth_req=req,
+            user_id=req["client_id"],
```

Comment:
client credentials is OAuth2 and we may consider the value of sub that's the resource owner, that in oidc is the user and in oauth2 is the client (RP)

Reply:
Don't be mislead by the name. idpy-oidc supports both oauth2 and oidc.
If we keep the triple (user_id, client_id, grant_id) then having user_id being equal to the client_id is much better than having None are client_id.
This is all about storing access tokens somewhere it's not about session management ?
Have to think about what it would take to allow a tuple with 2 values instead of 3 as key.

Reply:
> having user_id being equal to the client_id is much better than having None are client_id.

Intuitively, I also think this is better.

> Have to think about what it would take to allow a tuple with 2 values instead of 3 as key.

If we can get away from that by just using the client_id as the user_id then let's do it. I don't think we have (at the moment at least) any other case where there is no user involved, nor are there plans to have any other such flow. If this comes up we can dive into refactoring the code properly.

Reply:
I can live with that.
Even so, I'll probably have a look at how hard it would be to refactor.